lib/libxdp/libxdp.c

```diff
@@ -2491,6 +2493,9 @@ static struct xdp_multiprog *xdp_multiprog__from_id(__u32 id, __u32 hw_id,
 		err = PTR_ERR(mp);
 		goto err;
 	}
+	// duplicated fd/hw_fd already in prog, close originals
+	close(fd);
+	close(hw_fd);
```

Comment:
Hmm, actually, looking at the code again here, this function is not guaranteed to be called with both id and hw_id set, so here we *do* need a check for `fd > 0` before closing. Sorry for not realising that the first time around :(